gammapy/scripts/image_pipe.py

```diff
@@ -64,9 +67,12 @@ def make_counts(self, obs_id):
         obs = self.data_store.obs(obs_id=obs_id)
         events = obs.events.select_energy(self.energy_band)
         events = events.select_offset(self.offset_band)
-        counts_map.fill(events=events)
-        counts_map.data = counts_map.data.value
-        self.maps["counts"] = counts_map
+        if (len(events) != 0):
```

Comment:
I think ` _has_counts(obs_id)` is OK.

Checking counts is just a sum and should be blazing fast ... I think even if you do it multiple times per obs you won't even notice the slowdown really.